src/State/Inventory.cpp

```diff
@@ -308,6 +330,47 @@ void Inventory::onScrollDownButtonClick(Event::Mouse* event)
     if(inventory->canScrollDown())
     {
         inventory->scrollDown();
+        //enable/disable scroll buttons on downward scroll
+        if(!inventory->canScrollDown())
+        {
+            toggleScrollDownBtn(false);
+        }
+        if(inventory->canScrollUp())
+        {
+            toggleScrollUpBtn(true);
+        }
+    }
+}
+
+void Inventory::toggleScrollUpBtn(bool toggle)
+{
+    auto scrollUpBtn = dynamic_cast<UI::ImageButton*>(getUI("button_up"));
+    auto scrollUpBtnDisabled = dynamic_cast<UI::ImageButton*>(getUI("button_up_disabled"));
+    if(toggle)
```

Comment:
This can be just
    scrollUpBtnDisabled->setEnabled(!toggle);
    scrollUpBtn->setEnabled(toggle);